sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/predicates.scala

```diff
@@ -210,13 +210,25 @@ case class EqualNullSafe(left: Expression, right: Expression) extends BinaryComp
 case class LessThan(left: Expression, right: Expression) extends BinaryComparison {
   override def symbol: String = "<"
 
-  lazy val ordering: Ordering[Any] = {
+  lazy val orderings: Either[Ordering[Any], (Int, Ordering[Any])] = {
     if (left.dataType != right.dataType) {
       throw new TreeNodeException(this,
         s"Types do not match ${left.dataType} != ${right.dataType}")
     }
     left.dataType match {
-      case i: AtomicType => i.ordering.asInstanceOf[Ordering[Any]]
+      case i: AtomicType => Left(i.ordering.asInstanceOf[Ordering[Any]])
+      case s: StructType =>
+        var i = -1 
+        val f = s.fields.find { f =>
+         i += 1
```

Comment:
We shouldn't rely on side-effect of functions like this.  Can you just use `indexOf`?